lib/models/customer_info_wrapper.dart

```diff
@@ -1,67 +1,106 @@
-    /// it will point there. If there are no active subscriptions it will be null.
-    /// If there are multiple for different platforms, it will point to the device store.
-    String? managementURL,
-  }) = _CustomerInfo;
-
-  factory CustomerInfo.fromJson(Map<String, dynamic> json) =>
-      _$CustomerInfoFromJson(json);
+class CustomerInfo extends Equatable {
+  /// Entitlements attached to this customer info
+  final EntitlementInfos entitlements;
+
+  /// Map of skus to purchase dates
+  final Map<String, String?> allPurchaseDates;
+
+  /// Set of active subscription skus
+  final List<String> activeSubscriptions;
+
+  /// Set of purchased skus, active and inactive
+  final List<String> allPurchasedProductIdentifiers;
+
+  /// Returns all the non-subscription purchases a user has made.
+  /// The purchases are ordered by purchase date in ascending order.
+  final List<StoreTransaction> nonSubscriptionTransactions;
+
+  /// The date this user was first seen in RevenueCat.
+  final String firstSeen;
+
+  /// The original App User Id recorded for this user.
+  final String originalAppUserId;
+
+  /// Map of skus to expiration dates
+  final Map<String, String?> allExpirationDates;
+
+  /// Date when this info was requested
+  final String requestDate;
+
+  /// The latest expiration date of all purchased skus
+  final String? latestExpirationDate;
+
+  /// Returns the purchase date for the version of the application when the user bought the app.
+  /// Use this for grandfathering users when migrating to subscriptions.
+  final String? originalPurchaseDate;
+
+  /// Returns the version number for the version of the application when the
+  /// user bought the app. Use this for grandfathering users when migrating
+  /// to subscriptions.
+  ///
+  /// This corresponds to the value of CFBundleVersion (in iOS) in the
+  /// Info.plist file when the purchase was originally made. This is always null
+  /// in Android
+  final String? originalApplicationVersion;
+
+  /// URL to manage the active subscription of the user. If this user has an active iOS
+  /// subscription, this will point to the App Store, if the user has an active Play Store subscription
+  /// it will point there. If there are no active subscriptions it will be null.
+  /// If there are multiple for different platforms, it will point to the device store.
+  final String? managementURL;
+
+  const CustomerInfo(
+    this.entitlements,
+    this.allPurchaseDates,
+    this.activeSubscriptions,
+    this.allPurchasedProductIdentifiers,
+    this.nonSubscriptionTransactions,
+    this.firstSeen,
+    this.originalAppUserId,
+    this.allExpirationDates,
+    this.requestDate, {
+    this.latestExpirationDate,
+    this.originalPurchaseDate,
+    this.originalApplicationVersion,
+    this.managementURL,
+  });
+
+  factory CustomerInfo.fromJson(Map<String, dynamic> json) => CustomerInfo(
+      EntitlementInfos.fromJson(Map<String, dynamic>.from(json['entitlements'])),
```

Comment:
I guess it's error prone to have all this mappings done manually... I would like to write more tests for all these mappings before merging this.